File: 백준/Silver/1439. 뒤집기/뒤집기.java
import java.io.*;
import java.util.*;
import java.util.regex.*;;
public class Main{
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine().trim();

        Pattern pattern = Pattern.compile("(0+|1+)"); //정규 표현식으로 그룹
        Matcher matcher = pattern.matcher(s);

        Map<Integer, Integer> cnt = new HashMap<>();
        cnt.put(0,0);
        cnt.put(1,0);

        while(matcher.find()){
            String g = matcher.group(); // 매칭된 그룹 카운트하려는 while
            if(g.contains("1")){
                cnt.put(1,cnt.get(1)+1);
            }else{
                cnt.put(0,cnt.get(0)+1);
            }
        }
    System.out.println(Math.min(cnt.get(0),cnt.get(1)));
    }
}

File: 백준/Silver/1439. 뒤집기/뒤집기.py
import re
import sys
input = sys.stdin.readline

s = input().rstrip()
group = re.findall(r'(0+|1+)',s)
cnt = {0: 0, 1: 0}
for g in group:
    if '1' in g:
        cnt[1]+=1
    else:
        cnt[0]+=1

print(min(cnt.values()))
